nncf/quantization/algorithms/weight_compression/torch_backend.py

```diff
@@ -0,0 +1,239 @@
+
+        ndims = len(weight_node.layer_attributes.shape)
+        reduction_axis = None
+        if node_with_weight.metatype == om.PTEmbeddingMetatype:
+            reduction_axis = [1]
+        elif node_with_weight.metatype == om.PTLinearMetatype:
+            reduction_axis = [ndims - 1]
+        elif node_with_weight.metatype == om.PTMatMulMetatype:
+            if weight_port_id == 0:
+                reduction_axis = [ndims - 1]
+            elif weight_port_id == 1:
+                reduction_axis = [max(0, ndims - 2)]
+            reduction_axis = [max(0, reduction_axis)]
+        elif node_with_weight.metatype == om.PTAddmmMetatype:
+            if weight_port_id == 1:
+                reduction_axis = [ndims - 1]
+            elif weight_port_id == 2:
+                reduction_axis = [max(0, ndims - 2)]
+            reduction_axis = [max(0, reduction_axis)]
+        return reduction_axis
+
+    @staticmethod
+    def target_point(target_type: TargetType, target_node_name: str, port_id: int) -> PTTargetPoint:
+        if NNCFGraphNodeType.INPUT_NODE in target_node_name or target_type == TargetType.POST_LAYER_OPERATION:
+            port_id = None
+        if target_type in PTWeightCompressionAlgoBackend.TARGET_TYPE_TO_PT_INS_TYPE_MAP:
+            target_type = PTWeightCompressionAlgoBackend.TARGET_TYPE_TO_PT_INS_TYPE_MAP[target_type]
+        return PTTargetPoint(target_type, target_node_name, input_port_id=port_id)
+
+    @staticmethod
+    def raw_statistic_collector(num_samples: Optional[int] = None) -> TensorCollector:
+        return get_raw_stat_collector(num_samples)
+
+    @staticmethod
+    def get_activation_port_id(node: NNCFNode, graph: NNCFGraph) -> int:
+        activation_ports = []
+        for prev_node in graph.get_previous_nodes(node):
+            if prev_node.metatype in CONST_NOOP_METATYPES:
+                continue
+            edge = graph.get_edge(prev_node, node)
+            activation_ports.append(edge.input_port_id)
+        assert len(activation_ports) == 1
+        return activation_ports[0]
+
+    def get_weight(
+        self, node_with_weight: NNCFNode, weight_port_id: int, model: torch.nn.Module, graph: NNCFGraph
+    ) -> Tensor:
+        weight_node = get_weight_node(node_with_weight, weight_port_id, graph)
+        weight_name = weight_node.layer_attributes.name
+        module_name, weight_attr_name = split_weight_name(weight_name)
+        module = get_moodule_by_name(module_name, model)
+        weight = getattr(module, weight_attr_name)
+        if weight is None or not isinstance(weight, torch.nn.Parameter):
+            raise RuntimeError(f"Could not find a torch.nn.Parameter in the model by name {weight_name}.")
+
+        return Tensor(weight)
+
+    def transorm_model(
+        self, model: NNCFNetwork, graph: NNCFGraph, weight_compression_parameters: Iterable[WeightCompressionParameters]
+    ) -> NNCFNetwork:
+        # register storage for compression modules
```

Comment:
Is it possible to implement this function using the PTModelTransformer? Looks like some code is copy-pasted from it

Reply:
done